app/views/courses/ClassroomSettingsModal.coffee

```diff
@@ -65,3 +79,38 @@ module.exports = class ClassroomSettingsModal extends ModalView
       console.log 'e', e
       @$('#update-courses-btn').attr('disabled', false)
       noty { text: e.responseJSON?.message or e.responseText or 'Error!', type: 'error', timeout: 5000 }
+
+  onClickLinkGoogleClassroom: ->
+    application.gplusHandler.loadAPI({
+      success: =>
+        application.gplusHandler.connect({
+          scope: GoogleClassroomHandler.scopes
+          success: =>
+            $('#link-google-classroom-btn').text("Linking...")
+            $('#link-google-classroom-btn').attr('disabled', true)
```

Comment:
If we do it instantly on click, and if `loadAPI` or `connect` fails, then the button would remain disabled and the user would have to refresh the screen to move ahead, since we do not handle calling the error callback from the `gPlusHandler` currently (which ideally we should).
So for now, I think we can go ahead and disable the button on instant click, so that we avoid abuse of the button, knowing that an error in gPlusHandler would require users to refresh the screen.